File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    for target in targets:
        sum = 0
        for t in list(target):
            idx = 101
            for k in keymap:
                try:
                    idx = min(idx, k.index(t))
                except ValueError:
                    pass
            if idx == 101:
                sum = -1
                break
            else:
                sum += (idx+1)
        
        answer.append(sum)
        
    return answer

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

vector<int> solution(vector<string> keymap, vector<string> targets) {
    vector<int> answer;
    for(int i=0;i<targets.size();i++){
        int calc = 0;
        for(int j=0;j<targets[i].size();j++){
            int cnt = INT32_MAX;
            for(auto k : keymap){
                if(k.find(targets[i][j]) == -1) continue;
                cnt = min(cnt, (int) k.find(targets[i][j]));
                
            }
            if(cnt != INT32_MAX)
                calc += cnt + 1;
            else{
                calc = 0;
                break;
            }
        }
        if(calc != 0)
            answer.push_back(calc);
        else{
            answer.push_back(-1);
        }
            
    }
    return answer;
}